File: Problemas sobre RSQ y LIS/elis.py
from sys import stdin

def largest(numbers):
    n = len(numbers)
    lis = [1] * n

    for i in range(n - 1, -1, -1):
        for j in range(i + 1, n):
            if numbers[i] < numbers[j]:
                lis[i] = max(lis[i], 1 + lis[j])

    return max(lis)

input_data = stdin.read().strip().splitlines()

for i in range(0, len(input_data), 2):
    n = int(input_data[i])
    numbers = list(map(int, input_data[i + 1].split()))
    print(largest(numbers))

File: Problemas sobre RSQ y LIS/elis.js
const largest = (numbers) => {
    const n = numbers.length;
    const lis = new Array(n).fill(1);

    for (let i = n - 1; i >= 0; i--) {
        for (let j = i + 1; j < n; j++) {
            if (numbers[i] < numbers[j]) {
                lis[i] = Math.max(lis[i], 1 + lis[j]);
            }
        }
    }

    return Math.max(...lis);
}

const input_data = require('fs').readFileSync(0, 'utf-8').trim().split('\n');

for (let i = 0; i < input_data.length; i += 2) {
    const n = parseInt(input_data[i]);
    const numbers = input_data[i + 1].split(' ').map(Number);
    console.log(largest(numbers));
}